### DISCUSSION TITLE
MultiBlock.slice documentation

### DISCUSSION THREAD

**Discussion by athreya-jb** [pyvista/pyvista#3492] [Q&A]
Hey guys, I think I might have stumbled across a discrepancy in the documentation for MultiBlock datasets. I am very new to the VTK scene, and I'm learning by doing, so please go easy on me if I've missed something.

**The Issue**

I see in the [MultiBlock.slice filter documentation](https://docs.pyvista.org/api/core/_autosummary/pyvista.MultiBlock.slice.html) that the return type is `pyvista.PolyData`, but when I use this on my computer (Ubuntu 22.04 running Python3.8 using the Spyder IDE), I see that the filter returns `pyvista.MultiBlock` data (see attached image). Am I missing something?

I needed to save my slice data as PolyData (the downstream process involves using VTK.js, which apparently doesn' support multi-block data at all), so I manually do that by combining my MultiBlock  data to get Unstructured data, to which I then apply the `extract_surface` filter to finally get the PolyData object.

Screenshot from Spyder showing variable datatypes:
![spyder_vtm_to_vtp](https://user-images.githubusercontent.com/50865647/196680273-c6f7e258-d1da-4284-9d1f-48ce310d0c00.png)

My code is as follows:

```
import pyvista as pv

# Reading some sample MultiBlock data
vtm_data = pv.read("./data/volume_data/quantity_leaf_0.vtm")

# Creating a z-normal slice
slice_data = vtm_data.slice(normal=(0,0,1), origin=(0,0,0))

# Saving slice data as vtm
slice_data.save("./data/slices/slices_50_multi.vtm", binary=False)

# Using the extract_surface() filter on the MultiBlock slice to convert to PolyData
vtp_data = (slice_data.combine(merge_points=False)).extract_surface()
# Saving converted slice data as vtp
vtp_data.save("./data/slices/slice_50_poly.vtp", binary=False)

```

**Comment by MatthewFlamm** [comment]
Good catch.  Indeed, if a MultiBlock mesh is sliced, it slices each individual block and returns a new MultiBlock of slices of each.  This should be fixed in the documentation.

Your solution is probably the best: first slice, then combine and extract surface.

If you want one less step: you combine first, then you can eliminate the extract surface step  However, this operation will be more memory intensive, so I'd recommend your first approach unless you have a reason to use this one.

```py
slice_data = vtm_data.combine().slice(normal=(0, 0, 1))  # Get PolyData
```

**Reply by athreya-jb** [reply]
Thanks a lot for the additional info @MatthewFlamm! I think I will stick with the first method because, as you pointed out, memory usage is a crucial factor for my application.
